api/v1alpha1/kustomization_types.go

```diff
@@ -82,10 +83,9 @@ type KustomizationSpec struct {
 
 // WorkloadReference defines a reference to a Deployment, DaemonSet or StatefulSet.
 type WorkloadReference struct {
-	// Kind is the type of resource being referenced.
-	// +kubebuilder:validation:Enum=Deployment;DaemonSet;StatefulSet
+	// GroupKind is the type of resource being referenced.
 	// +required
-	Kind string `json:"kind"`
+	GroupKind metav1.GroupKind `json:"groupKind"`
```

Comment:
Need to update the docs with the api change.

Reply:
Let's make this the same with HPA and Flagger way of referencing objects:

```
// CrossNamespaceObjectReference contains enough information to let you locate the
// typed referenced object at cluster level
type CrossNamespaceObjectReference struct {
	// API version of the referent
	// +optional
	APIVersion string `json:"apiVersion,omitempty"`

	// Kind of the referent
	// +optional
	Kind string `json:"kind,omitempty"`

	// Name of the referent
	Name string `json:"name"`

	// Namespace of the referent
	// +optional
	Namespace string `json:"namespace,omitempty"`
}
```